independent-projects/tools/codestarts/README.adoc

```diff
@@ -33,210 +53,199 @@ Each codestart consists of:
-    fun hello() = "Hello foo"
-}
-----
+Those templates are able to use data which contains:
+
+* The `data` of the codestart to generate (specified in the `codestart.yml`)
+* A merge of the `shared-data` from the all the codestarts used to generate the project
+* The user input
+* Some dynamically generated data (e.g. `dependencies` and `test-dependencies`)
+
+=== README.md
 
-* kotlin/src/main/kotlin/org/acme/foo/Foo.kt
-* scala/src/main/scala/org/acme/foo/Foo.kt
+You may add a `README.md` or `README.tpl.qute.md` in the `base` directory, it will be appended to the others.
+So just add the info relative to your codestart.
 
-*NOTE* Just add `.tpl.qute` if you want it to be rendered with qute and use some data and simple logic
+=== application.properties
 
-*NOTE* You can use a `base` directory to add files to process for all languages: readme, configs, ...
+As a convention, you should always provide the Quarkus configuration as a yaml file (`src/test/resources/application.yml`).
 
+It is going to be:
 
-2. Add some configuration (if needed)
+* merged with the other codestarts configs
+* automatically converted to the selected config type (yaml or properties) at generation time depending on the selected extensions
 
-The `base/src/main/resources/application.yml` file is going to be merged with the other codestarts config and automatically converted to the selected config type (yaml or properties).
+=== index.html and cloud codestarts
 
-Note: if it's specific to a language, you can put the config in each languages folders
+Codestarts may provide a snippet for the generated index.html by adding this file:
 
-* base/src/main/resources/application.yml (using quarkus yaml config convention):
+base/src/main/resources/META-INF/resources/index.tpl.qute.html:
+[source,html]
+----
+<div class="example">
+    <div class="example-header">
+        <h4>{title}</h4>
+        <a href="{guide}" target="_blank" class="guide-link">Guide</a>
+    </div>
+    <div class="example-description">
+        {description}
+    </div>
+    <div class="example-paths">
+        {#for item in paths}
+        <a href="{item.path}" class="path-link" target="_blank">{item.method} {item.path}</a>
+        {/for}
+    </div>
+</div>
+----
+
+You also need to add this data to the `codestart.yml`:
+
+codestart.yml:
 [source,yaml]
 ----
-quarkus:
-  http:
-    port: 8081
+...
+language:
+  base:
+    data:
+      title: My example
+      description: |
+        <p>My example description.</p>
+        <p><b>A Quarkus catch phrase!</b></p>
+      paths:
+        - method: GET
+          path: /the-rest-path-of-my-example
+      guide: https://quarkus.io/guides/my-example
+...
 ----
 
-3. Create a readme section (if needed)
+=== Singleton example
 
-This will be appended to all different selected examples.
+If the codestart is a bad citizen and may have some compatibility issues, make it a `singleton-example`.
 
-* base/README.md
-[source,markdown]
-----
-# FOO Example
+Specificities:
```

Comment:
I would replace "Specificities" with something like "Be aware". But perhaps this list should be re-phrased to make it easier to read.

Reply:
Do you have something in mind @aloubyansky? I don't have a lot of inspiration to make it easier to read 😅

Reply:
I've tried something, let me know :)